bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/profile/DoorbirdProfileFactory.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.doorbird.internal.profile;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.profiles.Profile;
+import org.eclipse.smarthome.core.thing.profiles.ProfileCallback;
+import org.eclipse.smarthome.core.thing.profiles.ProfileContext;
+import org.eclipse.smarthome.core.thing.profiles.ProfileFactory;
+import org.eclipse.smarthome.core.thing.profiles.ProfileType;
+import org.eclipse.smarthome.core.thing.profiles.ProfileTypeProvider;
+import org.eclipse.smarthome.core.thing.profiles.ProfileTypeUID;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * A factory for Doorbird Profiles.
+ *
+ * @author Mark Hilbush - Initial contribution
+ *
+ */
+@NonNullByDefault
+@Component(service = { ProfileFactory.class, ProfileTypeProvider.class })
+public class DoorbirdProfileFactory implements ProfileFactory, ProfileTypeProvider {
+    private static final Set<ProfileType> SUPPORTED_PROFILE_TYPES = Collections
+            .singleton(DoorbirdProfiles.DOORBELL_COMMAND_TYPE);
+
+    private static final Set<ProfileTypeUID> SUPPORTED_PROFILE_TYPE_UIDS = Stream
+            .of(DoorbirdProfiles.DOORBELL_SWITCH_UID).collect(Collectors.toSet());
+
+    @Nullable
+    @Override
+    public Profile createProfile(ProfileTypeUID profileTypeUID, ProfileCallback callback, ProfileContext context) {
```

Comment:
Minor: wrong location of `@Nullable`:

```suggestion
    public @Nullable Profile createProfile(ProfileTypeUID profileTypeUID, ProfileCallback callback, ProfileContext context) {
```

Reply:
Done